src/wp-includes/fonts/class-wp-font-library.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Font Library class.
+ *
+ * This file contains the Font Library class definition.
+ *
+ * @package    WordPress
+ * @subpackage Font Library
+ * @since      6.4.0
+ */
+
+/**
+ * Font Library class.
+ *
+ * @since 6.4.0
+ */
+class WP_Font_Library {
+
+	/**
+	 * Fonts mime types allowed for each file type.
+	 * Each file type can have multiple mime types depending on the PHP version.
+	 * Currently wp_check_filetype_and_ext() only allows one mime type per file extension.
+	 */
+	const PHP_7_TTF_MIME_TYPE = PHP_VERSION_ID >= 70300 ? 'application/font-sfnt' : 'application/x-font-ttf';
+
+	const ALLOWED_FONT_MIME_TYPES = array(
+		'otf'   => 'font/otf',
+		'ttf'   => PHP_VERSION_ID >= 70400 ? 'font/sfnt' : self::PHP_7_TTF_MIME_TYPE,
+		'woff'  => PHP_VERSION_ID >= 80100 ? 'font/woff' : 'application/font-woff',
+		'woff2' => PHP_VERSION_ID >= 80100 ? 'font/woff2' : 'application/font-woff2',
+	);
```

Comment:
Condition logic here need comments.

Reply:
Is there a unit test for this?

Reply:
done here: https://github.com/WordPress/wordpress-develop/pull/5285/commits/04fe719b817b0c57e34442333a9c3744625cd2c4